composer/core/evaluator.py

```diff
@@ -59,6 +61,31 @@ def should_eval(state: State, event: Event):
             last_batch_seen = state.timestamp.batch
             return True
 
+        if eval_interval.unit == TimeUnit.DURATION:
+            assert state.max_duration is not None, 'max_duration should not be None'
+            if state.dataloader_len is None:
+                raise RuntimeError(
+                    f'Evaluation interval of type `dur` or {TimeUnit.DURATION} requires the dataloader to be sized.')
+            if state.max_duration.unit == TimeUnit.EPOCH and int(
+                    state.timestamp.batch) % math.ceil(state.max_duration.value * float(eval_interval) *
+                                                       state.dataloader_len) == 0 and event == Event.BATCH_END:
+                last_batch_seen = state.timestamp.batch
+                return True
+            elif state.max_duration.unit == TimeUnit.BATCH and int(state.timestamp.batch) % math.ceil(
+                    state.max_duration.value * eval_interval.value) == 0 and event == Event.BATCH_END:
+                last_batch_seen = state.timestamp.batch
+                return True
+            elif state.max_duration.unit == TimeUnit.SAMPLE and event == Event.BATCH_END:
+                # If last sample in batch is not evenly divisible by eval_interval, perform evaluation in next batch
+                if int(state.timestamp.batch) > 0:
+                    samples_in_a_batch = int(state.timestamp.sample) // int(state.timestamp.batch)
+                    if int(state.timestamp.sample) // math.ceil(state.max_duration.value * eval_interval) != int(
+                            state.timestamp.sample - samples_in_a_batch) // math.ceil(
+                                state.max_duration.value * eval_interval):
+                        last_batch_seen = state.timestamp.batch
+                        return True
+            elif state.max_duration.unit == TimeUnit.TOKEN and event == Event.BATCH_END:
+                raise ValueError(f'Evaluation interval of type `dur` is not support yet for max_duration as `tok`')
```

Comment:
```suggestion
                raise ValueError(f'Evaluation interval of type `dur` is not supported yet for max_duration as `tok`')
```

Reply:
Thanks! Done in https://github.com/mosaicml/composer/pull/1523/commits/55badd6111d6c79b705205444aa0d898df42b554.